website/docs/error-reference.md

```diff
@@ -53,6 +53,47 @@ and should not be reported.
 
 [#1993]: https://github.com/sorbet/sorbet/pull/1993
 
+## 3001
+
+Sorbet doesn’t support singleton definitions outside of the class itself:
+
+```rb
+class << MyClass # error: `class << EXPRESSION` is only supported for `class << self`
+  def foo
+    # ...
+  end
+end
+```
+
+The workaround is to move the definition inside the `MyClass` class itself:
+
+```rb
+class MyClass
+  class << self
+    def foo
+      # ...
+    end
+  end
+end
+```
+
+Sometimes, `EXPRESSION` is not a constant literal like in what follows:
+
+```rb
+class << some_variable
+  include Foo
+end
+```
+
+In this case, it is possible to directly call `include` on the the singleton
+class of `EXPRESSION`, but it should be done with **utmost caution**, as Sorbet
```

Comment:
The `utmost caution` part is copied from https://sorbet.org/docs/error-reference#4002

Reply:
Thanks for flagging this.

Nitpicking, I think we should revisit the language here and in the 4002 documentation, "far less accurate predictions" sounds vaguely like Sorbet is consulting a crystal ball -- maybe something like "make Sorbet's analyses far less accurate" would be better?  Regardless, this is followup material.